conversation_simulator/src/StateMachine.js

```diff
@@ -226,4 +230,32 @@ export default class {
     }
     return split
   }
+
+  // Marks that a specific answer was already used on a certain state
+  markSolution (state, answer) {
+    let stateName = this.getStateName(state)
+    if (!this.visitedStates[stateName]) {
+      this.visitedStates[stateName] = []
+    }
+    this.visitedStates[stateName].push(answer)
+  }
+
+  // Checks to see if a solution has already been used
+  checkIfSolutionUsed (state, answer) {
+    let stateName = this.getStateName(state)
+    if (this.visitedStates[stateName] && this.visitedStates[stateName].indexOf(answer) !== -1) {
+      return true
+    } else {
+      return false
+    }
+  }
+
+  // Finds a state object's name
+  getStateName (state) {
```

Comment:
StateMachine already has this function this.stateMachine.CurrentStateName()

Reply:
Noted, using that function now.